local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+"""Describes obstacles which the Sailbot must avoid: Boats and Land"""
+
+import math
+
+import numpy as np
+from custom_interfaces.msg import HelperAISShip
+from numpy.typing import NDArray
+from shapely.geometry import Point, Polygon
+
+from local_pathfinding.coord_systems import (
+    XY,
+    LatLon,
+    knots_to_kilometers_per_hour,
+    latlon_to_xy,
+    meters_to_km,
+)
+
+# Constants
+PROJ_TIME_NO_COLLISION = 3  # hours
+COLLISION_ZONE_SAFETY_BUFFER = 0.5  # km
+
+
+class Obstacle:
+
+    """
+    This Interface describes general obstacle objects which are
+    anything which the sailbot must avoid.
+
+    The choice was made to create an interface rather than a class as Boat and LandMass objects
+    may be very different objects which share an "avoidable" property and a position, but not much
+    else.
+
+    """
+
+    def __init__(
+        self,
+        reference: LatLon,
+        sailbot_position: LatLon,
+        sailbot_speed: float,
+    ):
+        """
+        Args:
+            -reference (LatLon): latitude and longitude of the next global waypoint
+            -sailbot_position (LatLon): lat and lon position of SailBot
+            -sailbot_speed (float): speed of the SailBot in km/h
```

Comment:
So sailbot_position will be published as an XY correct?
Will reference also be published as XY? 

I'm a little confused on what will be XY and what will be LatLon, I assumed I needed to use latlon_to_xy in my functions, because locations will be published in LatLon, passed in to functions as LatLon, and then converted to XY inside my functions.

Reply:
Hopefully we can keep the XY coordinate system internal to this file (not thought about anywhere else). The `self.sailbot_position` class attribute would be type XY. The reference XY coordinate would be (0, 0)